controllers/helmchartinstall_controller.go

```diff
@@ -47,8 +52,23 @@ func (r *HelmChartInstallReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	ctx := context.Background()
 	_ = r.Log.WithValues("helmchartinstall", req.NamespacedName)
 
-	// your logic here
-	i := &helmv1alpha1.HelmChartInstall{}
+	// TODO NOTE the group, version, and kind would normally come from the
+	// stack configuration, and would be part of the configuration for the render
+	// callback
+	//
+	// We grab the claim as an unstructured so that we can have the same code handle
+	// arbitrary claim types. The types will be erased by this point, so if a stack
+	// author wants to validate the schema of a claim, they can do it by putting a
+	// schema in the CRD of the claim type so that the claim's schema is validated
+	// at the time that the object is accepted by the api server.
+	i := &unstructured.Unstructured{}
+	i.SetGroupVersionKind(
+		schema.GroupVersionKind{
+			Group:   "helm.samples.stacks.crossplane.io",
+			Version: "v1alpha1",
+			Kind:    "HelmChartInstall",
+		},
+	)
 	if err := r.Client.Get(ctx, req.NamespacedName, i); err != nil {
```

Comment:
I would suggest not [embedding](https://golang.org/doc/effective_go.html#embedding) `Client` in `r`, especially given you're already calling `r.Client.Get` and not `r.Get`. 

When a type like `Client` is embedded in a struct its methods can be called as either `r.Client.Get` or `r.Get`. controller-runtime seems to prefer to embed the client, but I prefer to use embedded types sparingly, as they impact readability by making it appear as if the `Get` method is defined on `r`, when it is in fact defined by a type embedded in `r`. This gets especially confusing when there are multiple embedded types.